sympy/logic/tests/test_boolalg.py

```diff
@@ -828,3 +829,14 @@ def test_binary_symbols():
 
 def test_BooleanFunction_diff():
     assert And(x, y).diff(x) == Piecewise((0, Eq(y, False)), (1, True))
+
+
+def test_relational_simplification():
+    assert Or(x >= y, x < y).simplify() == S.true
```

Comment:
This can only be True if x and y are comparable. If either or both of x and y are imaginary then this cannot be evaluated.

Reply:
Yes, that is an inherent problem with all of this. The question is just if one should assume anything put in a comparison to be non-complex (I think there will be an exception if it is explicitly complex) or force the user to explicitly declare everything as real and check that for all simplifications.

I envision that x**2 < 0 may come up as an example where complex is valid, but I honestly think that x < I + 1 is as valid. In both cases the solutions are on a single line in the complex plane. We do not allow the second, so what is so special about the first?

Reply:
The first is comparing two real numbers when x is pure imaginary.

Reply:
The second also compares two real numbers, but along a different line in the complex space.

Reply:
I think the main argument though is that since these type of relations in general are not valid for complex numbers, it makes sense to assume that the inputs are real-valued.

Reply:
> second also compares two real numbers

Well, `1 + I` is not a real number, but yes, we can define it so that comparison of complex numbers is done by comparing the real parts. But that is not something that we allow (at present), choosing, rather, to raise an error for `1 < I` rather than False. I am not so savvy on all the implications. We try to be careful about assumptions and I don't see how we can, without user permission, say something is True or False when there are implicit assumptions being made. We have methods like `posify` and options like `force` in place. Perhaps simplify could get a flag `real=True` that would assume that vanilla symbols are real. (Actually, a more general `vanilla=dict(real=True, positive=...)` might be better where anything that is vanilla takes on the attributes passed so `vanilla=dict(postive=True)` could be used to make all vanilla symbols positive.)

Reply:
Yes, but when x is pure imaginary it is not a real number either.

I am not sure if we are fully understanding each other. My main argument is really that relations should imply real-value numbers as there is not a valid way to compare otherwise (except for Eq and Ne). Hence, making that assumption when simplifying should be OK. Right now it seems like one can sometimes use complex numbers, `x**2 < 0` implies a complex number (`Re(x) == 0` and `Im(x) != 0` for it to not be False) and is OK (since it not simplifies to False), `x < I + 1` (or more clearly, `x - I < 1` implies a complex number (`Im(x) == 1` leads to a real-valued comparison) and is not OK.

Maybe the way to go is to implement `relsimp` where it is clearly stated that it is assumed that the simplifications are valid for real-valued symbols only and that oo and -oo may be incorrectly handled (the `x(x+1) > x**2+x` case which in my approach will be False, not sure if this is the exactly correct formulation of where it is an issue). There is also the #15905 PR that can go into this function that has related assumptions. The only drawback I can see is that the code requires a bit of wrappers around it and must be written in a different way (or maybe one just renames* the current `_eval_simplify` to `_eval_relsimp` and writes a simple `relsimp` based on `simplify`).

* Extract the added parts to `_eval_relsimp` and keeps the previous ones.

Reply:
> Yes, but when x is pure imaginary it is not a real number either.
> 
> I am not sure if we are fully understanding each other

What I mean is that `x**2 < 0` becomes `-1 < 0` when x = I and that is a perfectly fine comparison, likewise for `x - I < 0` which becomes `0 < 0`. Both are legitimate comparisons.

> Maybe the way to go is to implement `relsimp`

My initial thought is that we already know how to deal with this in a sympythonic way: make the variables real. If someone is working with real variables they can jsut do `vars('a:z', real=True)` or if we make a `flavor` function then `flavor(real=True)` would change all vanilla symbols to real.

Reply:
The thing is that `x - I < 1` will throw an exception on you at the moment. So for consistency reasons `x**2 < 0` might just as well simplify to False. The obvious option is of course to not throw an exception for `x - I < 1`.

Of course a real check is an option. However, that require additional checking of every variable, which is maybe not a big deal though compared to the matching. I can try to include those both here and in the other PR.

Reply:
@smichr Try the following in your current sympy:

```
from sympy import symbols, Or
x, y = symbols('x y')
Or(x >= y, x < y)
```

Reply:
Hmm...it's already True without simplify. So it looks like we have a precedent to assume that vanilla symbols are real for purposes of comparison.

> that `x - I < 1` will throw an exception on you at the moment
In your PR? In master it doesn't throw an error.

Reply:
I was wrong there. `x - 1 < I` will though. In #15905 I have added a piece of code that will keep the imaginary constants in the exact position of the relation, so they won't simplify away in e.g. `x - I < y - I`. Not sure if I've pushed it though (doesn't look like it). Not sure if one can do something similar here, although I do not think it is an issue in the same way here.